src/lib/get-locale.js

```diff
@@ -0,0 +1,16 @@
+var Parser = require("accept-language-parser");
+var bestLang = require("bestlang");
+var langmap = require("langmap");
+var locales = require('../../public/locales.json');
+
+module.exports = function(acceptLang, locale) {
+  var langHeader = [];
+  var langArray = [];
+  if (locale && langmap[locale] && locales[locale]) {
+    return locale;
+  }
+  langHeader = Parser.parse(acceptLang);
```

Comment:
Otherwise, the guessed locale is not valid, so we need to return the header suggested locale, so a redirect can happen.